File: src/main/buttons/SaveFile.java
package main.buttons;

import main.Language;
import processing.core.PConstants;
import processing.data.JSONObject;

import java.time.LocalDateTime;

public class SaveFile extends LongButton {

    private String name;
    private JSONObject json;

    public SaveFile(String name, JSONObject json) {
        super();

        this.name = name;
        this.json = json;
    }

    public String getName() {
        return this.name;
    }

    public JSONObject getJson() {
        return this.json;
    }

    @Override
    protected void drawButton() {
        super.drawButton();

        main.textSize(20);
        main.textAlign(PConstants.LEFT, PConstants.CENTER);
        main.fill(255);
        main.text(this.name, this.getX() + 10, this.getY() + 28);

        main.textAlign(PConstants.RIGHT, PConstants.CENTER);
        main.fill(120);
        main.text(Language.getLanguageSelected().getLocalizedString("load game", "last modified") + ": " + this.json.getString("last modified"), this.getX() + WIDTH - 10, this.getY() + 28);
    }

    @Override
    public void clicked() {
        main.game.setSaveFile(this);
        main.switchRoom(main.game);
    }

    @Override
    public int compareTo(LongButton o) {
        return -LocalDateTime.parse(this.json.getString("last modified"), main.formatter).compareTo(LocalDateTime.parse(((SaveFile) o).json.getString("last modified"), main.formatter));
    }
}

File: src/main/buttons/SaveFile.kt
package main.buttons

import main.Language
import main.Main
import main.rooms.Game
import main.rooms.Game.gameLoaded
import main.rooms.Game.saveFile
import processing.core.PConstants
import processing.data.JSONObject
import java.time.LocalDateTime

class SaveFile(val name: String, val json: JSONObject) : LongButton() {
    override fun drawButton() {
        super.drawButton()
        Main.textSize(20f)
        Main.textAlign(PConstants.LEFT, PConstants.CENTER)
        Main.fill(255)
        Main.text(name, x + 10, y + 28)
        Main.textAlign(PConstants.RIGHT, PConstants.CENTER)
        Main.fill(120)
        Main.text(Language.languageSelected.getLocalizedString("load game", "last modified") + ": " + json.getString("last modified"), x + WIDTH - 10, y + 28)
    }

    override fun clicked() {
        saveFile = this
        gameLoaded = false
        Main.switchRoom(Game)
    }

    override fun compareTo(other: LongButton): Int {
        return -LocalDateTime.parse(json.getString("last modified"), Main.formatter).compareTo(LocalDateTime.parse((other as SaveFile).json.getString("last modified"), Main.formatter))
    }
}